ui/mirage/factories/agent.js

```diff
@@ -8,20 +8,51 @@ const AGENT_STATUSES = ['alive', 'leaving', 'left', 'failed'];
 
 export default Factory.extend({
   id: i => (i / 100 >= 1 ? `${UUIDS[i]}-${i}` : UUIDS[i]),
-  name: () => `nomad@${faker.random.boolean() ? faker.internet.ip() : faker.internet.ipv6()}`,
+  config: makeConfig,
 
-  status: () => faker.helpers.randomize(AGENT_STATUSES),
-  serfPort: () => faker.random.number({ min: 4000, max: 4999 }),
-
-  address() {
-    return this.name.split('@')[1];
-  },
-
-  tags() {
-    const rpcPortCandidate = faker.random.number({ min: 4000, max: 4999 });
+  member() {
```

Comment:
is there anything stopping us from simplifying this to:

```
member: {
  x,
  y,
  z
}
```

Reply:
This attribute has dynamic values, so I guess that's why it needs to be a function. Although I tried it as an object and it worked as well, so I guess it's up to you 😄